sdk/Sdk/Tasks/ZipDeploy/ZipDeployTask.cs

```diff
@@ -89,6 +95,12 @@ internal async System.Threading.Tasks.Task<bool> ZipDeployAsync(string zipToPubl
                 return false;
             }
 
+            // publish endpoint differs when using a blob storage container
+            var publishUriPath = useBlobContainerDeploy ? "publish" : "zipdeploy";
```

Comment:
Apart from updating the /publish endpoint, we also need to pass a required parameter for remote build and deployer parameter for better telemetry: 
`/publish?Deployer=VisualStudio&RemoteBuild=false`

Reply:
This task could be executed outside of VS. This could be set using the agent provided if needed.

@patelchandni how is the deployer information being used?

Reply:
@fabiocav - Mainly `RemoteBuild=false` is a required parameter for new API. 
And `Deployer=VisualStudio` is an optional parameter for better deployment telemetry internally. Agent is a lot of times very cryptic whereas we have provided an option to give straight forward value for deployer. So, Az CLI, VS Code, DevOps, and GitHub Action are all now passing this parameter.

Reply:
I'm OK with logic that resolves a friendly deployer based on the agent. Not proposing the use of the literal agent, though, if using for telemetry, using that information would not be a bad idea as it would give you the ability to track behaviors across different versions of the extension.

The key thing is that we shouldn't assume VS in this path as that wouldn't be accurate.

Reply:
Done. 
Modified the URLs as:
- ZipDeploy : `<publishUrl>/api/zipdeploy?isAsync=true`
- OneDeploy (thru ZipDeploy): `<publishUrl>/api/publish?RemoteBuild=false`

Reply:
Thank you @anvillan. To confirm, this OneDeploy option is only support for function apps on Flex consumption, right?